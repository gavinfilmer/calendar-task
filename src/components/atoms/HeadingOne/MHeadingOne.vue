<template>
  <component :is="tag" v-if="linebreak" class="m-heading-one">
    <span
      v-for="paragraph in paragraphs"
      :key="paragraph"
      class="m-heading-one--line"
      :class="[{ 'm-heading-one--title-dot': addedDot }]"
      v-html="paragraph"
    />
  </component>
  <component
    :is="tag"
    v-else
    class="m-heading-one"
    :class="[{ 'm-heading-one--title-dot': addedDot }]"
    v-html="parsedHeading"
  />
</template>

<script>
export default {
  name: 'MHeadingOne',
  props: {
    tag: {
      type: String,
      required: false,
      default: 'h1',
    },
    heading: {
      type: String,
      required: false,
      default: 'Bought By Many',
    },
    addedDot: {
      type: Boolean,
      required: false,
      default: true,
    },
    linebreak: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
  computed: {
    parsedHeading() {
      const parsedDot = this.heading.endsWith('.') && this.addedDot
      return parsedDot ? this.heading.slice(0, -1) : this.heading
    },
    paragraphs() {
      /* Split at full stop to create array for multi-paragraph headlines */
      return this.heading
        .split('. ')
        .map(h => h.replace('.', ''))
        .map(h => (!this.addedDot ? h + '.' : h))
    },
  },
}
</script>

<style lang="scss">
@import '../../../sass/framework/framework.scss';

.m-heading-one {
  &--title-dot {
    @include title-dot;
  }
  &--line {
    display: block;
  }
}
</style>
