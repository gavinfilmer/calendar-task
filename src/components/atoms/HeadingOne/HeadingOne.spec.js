import { mount, shallowMount } from '@vue/test-utils'
import MHeadingOne from './MHeadingOne.vue'

describe('HeadingOne', () => {
  it('renders correctly', () => {
    const wrapper = mount(MHeadingOne)
    expect(wrapper).toMatchSnapshot()
  })

  it('should default to "Bought By Many" heading', () => {
    const wrapper = shallowMount(MHeadingOne)
    expect(wrapper.find('h1').text()).toBe('Bought By Many')
  })

  test('renders correctly when heading string prop is passed', () => {
    const wrapper = shallowMount(MHeadingOne, {
      propsData: {
        heading: 'Passed Header String',
      },
    })
    expect(wrapper.find('h1').text()).toBe('Passed Header String')
  })
})
