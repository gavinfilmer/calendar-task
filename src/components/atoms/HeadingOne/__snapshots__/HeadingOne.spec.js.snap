// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingOne renders correctly 1`] = `<h1 class="m-heading-one m-heading-one--title-dot">Bought By Many</h1>`;
